File: api/ts/models/course.ts
import { Schema, connection, Types } from 'mongoose';

export interface Course {
  _id: Types.ObjectId;
  course_number: string;
  subject_prefix: string;
  title: string;
  description: string;
  school: string;
  credit_hours: string;
  class_level: string;
  activity_type: string;
  grading: string;
  internal_course_number: string;
  prerequisites: object;
  corequisites: object;
  co_or_pre_requisites: object;
  sections: Types.ObjectId[];
  lecture_contact_hours: string;
  laboratory_contact_hours: string;
  offering_frequency: string;
  attributes: object;
}

export const CourseSchema = new Schema<Course>({
  _id: { type: Types.ObjectId, required: true },
  course_number: { type: String, required: true },
  subject_prefix: { type: String, required: true },
  title: { type: String, required: true },
  description: { type: String, required: true },
  school: { type: String, required: true },
  credit_hours: { type: String, required: true },
  class_level: { type: String, required: true },
  activity_type: { type: String, required: true },
  grading: { type: String, required: true },
  internal_course_number: { type: String, required: true },
  prerequisites: { type: Object, required: true },
  corequisites: { type: Object, required: true },
  co_or_pre_requisites: { type: Object, required: true },
  sections: { type: [Types.ObjectId], required: true },
  lecture_contact_hours: { type: String, required: false },
  laboratory_contact_hours: { type: String, required: false },
  offering_frequency: { type: String, required: false },
  attributes: { type: Object, required: true },
});

const courseDB = connection.useDb('combinedDB');
export const CourseModel = courseDB.model<Course>('course', CourseSchema);

File: api-graphql/graph/model/course.go
package model

type Course struct {
	ID                     string                 `json:"_id"`
	CourseNumber           string                 `json:"course_number"`
	SubjectPrefix          string                 `json:"subject_prefix"`
	Title                  string                 `json:"title"`
	Description            string                 `json:"description"`
	School                 string                 `json:"school"`
	CreditHours            string                 `json:"credit_hours"`
	ClassLevel             string                 `json:"class_level"`
	ActivityType           string                 `json:"activity_type"`
	Grading                string                 `json:"grading"`
	InternalCourseNumber   string                 `json:"internal_course_number"`
	Prerequisites          *CollectionRequirement `json:"prerequisites"`
	Corequisites           *CollectionRequirement `json:"corequisites"`
	CoOrPreRequisites      *CollectionRequirement `json:"co_or_pre_requisites"`
	Sections               []*Section             `json:"sections"`
	LectureContactHours    string                 `json:"lecture_contact_hours"`
	LaboratoryContactHours string                 `json:"laboratory_contact_hours"`
	OfferingFrequency      string                 `json:"offering_frequency"`
}

func (Course) IsOutcome() {}
